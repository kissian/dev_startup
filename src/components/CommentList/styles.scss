
.comments {
  display: flex;

  .list-unstyled {
      
      li{
        font-size: 12px; line-height: 14px;
        .hidden-user {
          font-weight: 700;
          font-size: 13px;
          color: black;
        }
        .hidden-text{
          padding: 7px;
          display: inline-block;
          text-overflow: ellipsis;
          word-wrap: break-word;
          white-space: pre-wrap;
          color: #999;
          overflow: hidden;

        }
        .hidden-box {
          padding: 10px;
          border-radius: 2px;
          background-color: #f9f9f9;
          color: #111;
        }
      }
  }


}