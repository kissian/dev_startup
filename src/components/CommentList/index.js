import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-bootstrap';
import Moment from 'components/Moment';
import './styles.scss';

const CommentList = ({
    comments,
}) => (
    <Col lg={12} className="comments">
        <ul className="list-unstyled">
            { comments.map(({
                id, name, comment, date,
            }) => (
                <li key={id}>
                    <p className="hidden-user">
                        {name}
                    </p>
                    <span className="hidden-text">
                        {comment}
                    </span>
                    <div className="hidden-box">
                        <Moment date={date} />
                    </div>
                </li>
            ))
            }
        </ul>
    </Col>
);
CommentList.propTypes = {
    comments: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        comment: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
    })).isRequired,
};

export default CommentList;
