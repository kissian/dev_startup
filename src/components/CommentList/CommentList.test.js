import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import CommentList from './index';
//comments
//id, name, comment, date,

describe('<CommentList />', () => {
  let comments = [
    {
      "id": "0",
      "name": "seb",
      "comment": "hellow comment 1",
      "date": "Thu Jul 26 2018 18:13:31 GMT+0900 (한국 표준시)"
    },
    {
      "id": "1",
      "name": "ian",
      "comment": "hellow comment 2",
      "date": "Thu Jul 26 2018 18:14:31 GMT+0900 (한국 표준시)"
    }
  ]
  const component = shallow(
    <CommentList 
      comments={comments}
    />
  )

  beforeEach(() => {component});

  it("+++ CommentList snapshot", () => {
    expect(component).toMatchSnapshot();
  })
})