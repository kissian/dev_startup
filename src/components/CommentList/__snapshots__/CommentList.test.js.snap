// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList /> +++ CommentList snapshot 1`] = `
<Col
  bsClass="col"
  className="comments"
  componentClass="div"
  lg={12}
>
  <ul
    className="list-unstyled"
  >
    <li
      key="0"
    >
      <p
        className="hidden-user"
      >
        seb
      </p>
      <span
        className="hidden-text"
      >
        hellow comment 1
      </span>
      <div
        className="hidden-box"
      >
        Thu Jul 26 2018 18:13:31 GMT+0900 (한국 표준시)
      </div>
    </li>
    <li
      key="1"
    >
      <p
        className="hidden-user"
      >
        ian
      </p>
      <span
        className="hidden-text"
      >
        hellow comment 2
      </span>
      <div
        className="hidden-box"
      >
        Thu Jul 26 2018 18:14:31 GMT+0900 (한국 표준시)
      </div>
    </li>
  </ul>
</Col>
`;
